ci/util.py

```diff
@@ -262,24 +302,44 @@ def _count_elements(value, count=0, max_elements_count=100000):
 
 def lint_yaml_file(path, linter_config: dict={'extends': 'relaxed'}):
     existing_file(path)
-    cfg = yamllint.config.YamlLintConfig(yaml.dump(linter_config))
     info(f'linting YAML file: {path}')
-    worst_level = 0
+
     with open(path) as f:
-        for problem in yamllint.linter.run(f, conf=cfg, filepath=path):
-            # problem is of type yamllint.linter.LintProblem
-            level = problem.level
-            if isinstance(problem.level, str):
-                level = yamllint.linter.PROBLEM_LEVELS[problem.level]
-
-            if level == yamllint.linter.PROBLEM_LEVELS['warning']:
-                warning(str(problem))
-            elif level > yamllint.linter.PROBLEM_LEVELS['warning']:
-                error(str(problem))
-
-            worst_level = max(worst_level, level)
-    if worst_level >= yamllint.linter.PROBLEM_LEVELS['error']:
-        fail('found errors whilst linting (see above)')
+        linting_result = _lint_yaml(f.read(), linter_config)
+
+    if not linting_result.has_problems():
+        return
+
+    for level, problem in linting_result.problems_iter():
+        if level < yamllint.linter.PROBLEM_LEVELS['error']:
+            warning(problem)
+        else:
+            error(problem)
+
+    if linting_result.max_level() >= yamllint.linter.PROBLEM_LEVELS['error']:
+        raise LintingError('Found some Errors while linting. See above.')
+
+
+def _lint_yaml(input, config):
+    cfg = yamllint.config.YamlLintConfig(yaml.dump(config))
+    return LintingResult(yamllint.linter.run(input=input, conf=cfg))
+
+
+def lint_yaml(input, config={'extends': 'relaxed'}):
+
+    linting_result = _lint_yaml(input=input, config=config)
+
+    if not linting_result.has_problems():
+        return
+
+    if linting_result.max_level() <= yamllint.linter.PROBLEM_LEVELS['warning']:
+        warning('Found some problems while linting:\n\t' + '\n\t'.join(
```

Comment:
nit: rather use f-str? also, rather deduplicate the msg, please
also: why not add `__str__` to linting_result. would make a lot more sense to me than the `problems_iter` method.